assets/js/main.js

```diff
@@ -0,0 +1,25 @@
+let oldHash = "";
+
+function checkHashChanged() {
+    if (window.location.hash != oldHash) {
+        console.log("Hash changed!");
+        oldHash = window.location.hash;
+
+        let foundElement = null;
+
+        for (let topic of document.getElementsByClassName("topic")) {
+            if (topic.getAttribute("id") === oldHash.substring(1)) {
+                topic.setAttribute("open", "");
+                foundElement = topic;
+            } else {
+                topic.removeAttribute("open");
+            }
+        }
+
+        if (foundElement) {
+            foundElement.scrollIntoView();
+        }
+    }
+}
+
+setInterval(checkHashChanged, 100);
```

Comment:
Please don't busy wait. We have the [haschange event](https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event) nowadays.